wikimedia/wikipedia-ios
# Protected page warning

**Phabricator:**  https://phabricator.wikimedia.org/T313772

### Notes
* This PR adds a warning panel on protected pages when the user doesn't have the necessary rights to perform and edit
* The same logic from the block messages was used: the protected information is received upon clicking on the editing button and loading the wikitext. If the user can't edit, the editor will be displayed as read-only, and a modal with the protection information will be shown.
* Edit notices are not displayed as a modal in this case.

### Test Steps
1. Go to a protected or semi-protected page using an account that does not have the correct permission to edit. Confirm that the modal is shown with the protection info. 
2. Note that, at this moment, the icon is not correct. It will be updated soon.
